### PR TITLE
Adding Transfer as a destination

### PR DESCRIPTION
**Pull request by Tang8330**
## Changes

This PR adds the capability to use Transfer as a destination as opposed to Kafka. This provides benefits during snapshots:

1. No Kafka growth
2. We are using `APPEND` and not `MERGE`, so it'll be faster to load the data

When specifying Transfer as the destination, it'll call all the same Transfer methods under the hood, as well as provide consistent telemetry on data flushing and processing

### CODE REVIEW

File: lib/transfer/writer.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+func (w *Writer) WriteRawMessages(_ context.Context, rawMsgs []lib.RawMessage) error {
+	if len(rawMsgs) == 0 {
+		return nil
+	}
+
+	var events []event.Event
+	for _, rawMsg := range rawMsgs {
+		evt := rawMsg.Event()
+		if payload, ok := evt.(*util.SchemaEventPayload); ok {
```
Comment (Tang8330) [inline lib/transfer/writer.go:71]: We don't need to JSON marshal and unmarshal for MongoDB?

File: config/config.go [hunk 2]
```diff
@@ -139,6 +142,31 @@ func (s *Settings) Validate() error {
 		if err := s.Kafka.Validate(); err != nil {
 			return fmt.Errorf("kafka validation failed: %w", err)
 		}
+	case DestinationTransfer:
+		if s.Transfer == nil {
+			return fmt.Errorf("transfer config is nil")
+		}
+
+		toipicConfigs, err := s.Transfer.TopicConfigs()
```
Comment (Tang8330) [inline config/config.go:150]: typo

File: writers/transfer/writer.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+		// The Mongo after payload is a string so no need to sanitize it.
+	default:
+		return event.Event{}, fmt.Errorf("unsupported event type: %T", evt)
+	}
+
+	return event.ToMemoryEvent(evt, message.PartitionKey(), topicConfig, config.Replication), nil
+}
+
+type Writer struct {
+	// TODO: How do we generate this
```
Comment (Tang8330) [inline writers/transfer/writer.go:56]: Remove TODO?

File: writers/transfer/writer.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+func (w *Writer) getTableData() (string, *models.TableData) {
+	tableData := w.inMemDB.TableData()
+	if len(tableData) != 1 {
+		panic("expected exactly one table")
+	}
+	for k, v := range tableData {
+		return k, v
+	}
+	panic("can not happen")
```
Comment (Tang8330) [inline writers/transfer/writer.go:131]: Throw an error instead?

**Review by nathan-artie — APPROVED** [review]

**Pull request merged**